python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -820,7 +820,7 @@ def _apply_filter_to_query(
                 > datetime.utcfromtimestamp(event_records_filter.after_timestamp)
             )
 
-        if event_records_filter.storage_ids:
+        if event_records_filter.storage_ids is not None:
```

Comment:
I think + the change above is risky - version skew will cause any old `EventRecordsFilter` to trigger this clause and I assume get 0 results back.

Whats the context on this? Am I misunderstanding that this would cause the query to return 0 results? Can we not detect this condition at the callsite and just skip asking the DB to give us back an empty list?

Reply:
Yeah, this value in an old event records filter would be an empty list and thus return 0 results. My motivation for making this change was to fix the existing behavior that passing in an empty list results in no filtering, since I think the expected behavior is to return no results.

I'm a bit confused about how version skew would occur. In OSS it has to be imported from the dagster library, which should ensure that in this version and onwards the `get_event_records` call handles the None vs empty list distinction correctly. Or is it possible version skew can occur on cloud?

Reply:
Regardless, I'm fine with just detecting the condition at the callsite

Reply:
ahhh I see, thanks for explaining. So it seems like in order to support these cases, in cloud we would still have to do no filtering when an empty list is passed in